state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2044,8 +2053,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		origin: DepositOrigin<T, I>,
 	) -> DepositAction<T, I> {
 		match action.clone() {
-			ChannelAction::LiquidityProvision { lp_account, .. } => {
+			ChannelAction::LiquidityProvision { lp_account, additional_action, .. } => {
 				T::Balance::credit_account(&lp_account, asset.into(), amount_after_fees.into());
+
+				// TODO execute additional_action
```

Comment:
When we dispatch the account creation action (ie. when we receive the deposit into the channel), we want to:

- Estimate the swap input amount for 5 FLIP (or whatever our MIN_FUNDING is in the funding pallet)
- Take this off the actual deposit amount (like we do for ingress fees)
- Initiate a swap into FLIP with this much of the input
- Credit the target account with 0.01 FLIP: this should be enough to get started, pending completion of the swap.
- When the swap completes, credit the target account with the net amount (ie. swap_output_amount - 0.01 FLIP).